### DISCUSSION TITLE
How to increase performance of Combobox?

### DISCUSSION THREAD

**Discussion by patrickyan** [tailwindlabs/headlessui#1993] [Help]
Combobox is useful for filtering a long select list of options. However, long lists seem to make Combobox laggy. With a country select list (250 items, fairly common usage), it takes about 1 second for the list to actually pop up.

**Comment by thomas-sc** [comment]
I have the same problem with a list of several hundred taxonomy items. Combobox is unusable for this, although in my opinion that is the purpose of a combobox with autocomplete. For a short list of 10 items I can use a conventional select list. 
React-Select works fine with 1700 items, but it is not headless and much too complicated to style.

**Comment by patrickyan** [comment]
The solution right now is to use [@tanstack/react-virtual](https://tanstack.com/virtual/v3) to implement a virtualized list for <ComboBox.Options>
